ios/Sources/AppFeature/AppFeature.swift

```diff
@@ -6,43 +6,65 @@ import MediaFeature
 import Model
 import Repository
 
-public enum AppState: Equatable {
-    case needToInitialize
-    case initialized(AppTabState)
-    case errorOccurred
+public struct AppState: Equatable {
+    public var type: AppStateType
+    public var appTabState: AppTabState
+    public var language: Lang
 
-    public init() {
-        self = .needToInitialize
+    public init(type: AppStateType, language: Lang, feedContents: [FeedContent] = []) {
+        self.type = type
+        self.language = language
+        self.appTabState = AppTabState(feedContents: feedContents, language: language)
     }
 }
 
+public enum AppStateType: Equatable {
+    case needToInitialize
+    case initialized
+    case errorOccurred
+}
+
 public enum AppAction {
+    case onAppear
     case refresh
     case needRefresh
     case refreshResponse(Result<[FeedContent], KotlinError>)
     case appTab(AppTabAction)
+    case currentLanguage(Result<Lang?, KotlinError>)
 }
 
 public let appReducer = Reducer<AppState, AppAction, AppEnvironment>.combine(
     appTabReducer.pullback(
-        state: /AppState.initialized,
+        state: \.appTabState,
         action: /AppAction.appTab,
         environment: { $0 }
     ),
     .init { state, action, environment in
         switch action {
+        case .onAppear:
+            return environment.languageRepository
+                .currentLanguage()
+                .receive(on: DispatchQueue.main)
+                .catchToEffect()
+                .map(AppAction.currentLanguage)
```

Comment:
This case is not needed.
Please implement this with refresh timing 🙏🏼 

I think https://github.com/DroidKaigi/conference-app-2021/blob/main/ios/Sources/AboutFeature/AboutFeature.swift#L50-L53 these lines are useful for you 👍🏼

Reply:
I fixed it :+1: